.github/workflows/drepecated-functions-checker.yml

```diff
@@ -0,0 +1,67 @@
+name: "Check Deprecations"
+
+on:
+  pull_request:
+    types: [opened, synchronize, labeled, unlabeled]
+    # Runs on PRs to main
+    branches:
+      - main
+
+jobs:
+  deprecations-check:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 0 # by default the checkout action doesn't checkout all branches
+      - name: Set up Go
+        uses: actions/setup-go@4d34df0c2316fe8122ab82dc22947d607c0c91f9 #v4
+        with:
+          go-version-file: ./.go-version
+          cache: true
+      - name: Check deprecations for files in diff
+        run: |  
+          # Staticcheck uses static analysis to finds bugs and performance issues, offers simplifications, 
+          # and enforces style rules.
+          # Here, it is used to check if a deprecated function, variable, constant or field is used.
+
+          echo "Installing staticcheck"
+          go install honnef.co/go/tools/cmd/staticcheck@2023.1.2 #v0.4.2
+
+          # Run staticcheck 
+          echo "Performing Deprecations Check: Running Staticcheck"
+          staticcheck ./... | grep deprecated > staticcheckOutput.txt
```

Comment:
I am not sure if we need to add the correct build tags for enterprise. This would run similar to how changelog or security scanner would run. Running this command at the base of vault-enterprise directory (when this runs in enterprise, it would checkout enterprise code) and would give us the results for that code base. Is my understanding incorrect?

Reply:
Staticcheck does seem to respect build tags, so in order to run this check over the enterprise code, we'll need to make sure to include build tags: https://staticcheck.io/docs/running-staticcheck/cli/build-tags/.

There might be other places in the Makefile where it we're running `go vet` or other linters and not passing in build tags. We should make a ticket to take a look at that and see where build tags are missing.